contracts/modules/PermissionManager/GeneralPermissionManager.sol

```diff
@@ -81,18 +88,144 @@ contract GeneralPermissionManager is IPermissionManager, Module {
         emit ChangePermission(_delegate, _module, _perm, _valid, now);
         return true;
     }
-
+    
     /**
-    * @notice Use to get the details of the delegate
+    * @notice Use to change one or more permissions for a single delegate at once
     * @param _delegate Ethereum address of the delegate
-    * @return Details of the delegate
+    * @param _multiModules Multiple module matching the multiperms, needs to be same length
+    * @param _multiPerms Multiple permission flag needs to be changed
+    * @return nothing
+    */
+    function changePermissionMulti(
+        address _delegate,
+        address[] _multiModules,
+        bytes32[] _multiPerms
+    )
+    external
+    withPerm(CHANGE_PERMISSION)
+    {
+        require(delegateDetails[_delegate] != bytes32(0), "Delegate details not set");
+        require(_multiModules.length != 0 && _multiPerms.length !=0);
+        require(_multiModules.length == _multiPerms.length, "modules and permission length are not same");
+        
+        for(uint8 i=0;i<_multiPerms.length;i++){
+            bool _currentPerm = !perms[_multiModules[i]][_delegate][_multiPerms[i]];
```

Comment:
instead of duplicating the logic, for each entry call `changePermission` @comeonbuddy